lib/queue.rb

```diff
@@ -1,31 +1,73 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(50)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # ORIGNAL WAY DONE IN CLASS 
+    # if @front == -1 && @back == -1
+    #   @front = 0
+    #   @back = -1
+    # end
+
+    # if @front == @back 
+    #   raise ArgumentError.new('Queue is full and theres no waiting list')
+    # end 
+
+    # @store[@back] = element 
+    # @back = (@back + 1) % @store.length   
+
+    if @back == @store.size - 1 # if queue is full 
```

Comment:
Except that you should be able to wrap around to the front of the array!  That's why it's circular.